doc/source/notebooks/manual/understanding/tf_graphs_and_sessions.ipynb

```diff
@@ -4,9 +4,169 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "Handling TensorFlow graphs and Sessions\n",
+    "Handling TensorFlow graphs and sessions\n",
     "--\n",
-    "\n"
+    "\n",
+    "##### Check [Tips and Tricks notebook](../tips_and_tricks.ipynb) for more examples about graph and sessions."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import gpflow\n",
+    "import tensorflow as tf"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "\n",
+    "TensorFlow gives you a default graph, which you fill with tensors and operations - nodes and edges in the graph respectively. You can find details [here](https://www.tensorflow.org/guide/graphs#building_a_tfgraph) on how to change the default graph to another one or exploit multiple graphs.\n",
+    "\n",
+    "TensorFlow graph is a representation of your computation, to execute your code you need TensorFlow [session](https://www.tensorflow.org/guide/graphs#executing_a_graph_in_a_tfsession). For example, you can think of graphs and sessions as binary sources and actual command running it in  a terminal. Normally, TensorFlow doesn't provide the default session, although GPflow creates default session and you can get it via:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "session = gpflow.get_default_session()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "To change GPflows' default session:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "gpflow.reset_default_session()\n",
+    "assert session is not gpflow.get_default_session()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "You can manupulate sessions manually, but you have to make them default for GPflow:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "                                     class prior transform  trainable shape  \\\n",
+      "SquaredExponential/lengthscales  Parameter  None       +ve       True    ()   \n",
+      "SquaredExponential/variance      Parameter  None       +ve       True    ()   \n",
+      "\n",
+      "                                 fixed_shape value  \n",
+      "SquaredExponential/lengthscales         True   1.0  \n",
+      "SquaredExponential/variance             True   1.0  \n"
+     ]
+    }
+   ],
+   "source": [
+    "with tf.Session() as session:\n",
+    "    k = gpflow.kernels.RBF(1)\n",
+    "    print(k)"
```

Comment:
I think the formatting looks nicer if you don't wrap it in a `print()`

Reply:
it will not print k then